pkg/ee/metering/report_config_handler.go

```diff
@@ -112,6 +118,14 @@ func (m updateReportConfigurationReq) Validate() error {
 		}
 	}
 
+	if m.Body.Interval < 0 { // not 1 because 0 is a defaulted value
+		return k8cerrors.NewBadRequest("interval value cannot be smaller than 1.")
```

Comment:
The error message doesn't match the condition.

Reply:
I had a problem with this one. Technically `interval` shouldn't be smaller than `1`, but as this is an optional parameter it defaults to `0` in case it's not provided in the request, hence I left the comment.

I can match the error message with condition but then we would have different validation rules for `POST` and `PUT` of the same model field.

Alternatively, I could put it differently like this:
```suggestion
	if m.Body.Interval != 0 { // checking if interval value was not defaulted
		if m.Body.Interval < 1 {
			return k8cerrors.NewBadRequest("interval value cannot be smaller than 1.")
```

Reply:
Alternatively, use a `*int` (or even `*uint32`), then you can easily distinguish between "not set" and 0. :) Wouldn't be the first time int pointers are used in Kubernetes.